### PR TITLE
docker modules: various adjustments

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Various adjustments for docker modules:
 1. Move docker_* `module_utils` into subpackage;
 2. Drop Python 2.6 support (will no longer be available for Ansible 2.8).

##### ISSUE TYPE
- Bugfix Pull Request
- Feature Pull Request

##### COMPONENT NAME
docker_compose
docker_config
docker_container
docker_container_facts
docker_host_facts
docker_image
docker_image_facts
docker_login
docker_network
docker_network_facts
docker_node
docker_node_facts
docker_prune
docker_swarm
docker_volume
docker_volume_facts

### CODE REVIEW

File: lib/ansible/module_utils/docker/docker_swarm.py [hunk 1]
```diff
@@ -7,11 +7,11 @@
 try:
     from docker.errors import APIError
 except ImportError:
-    # missing docker-py handled in ansible.module_utils.docker_common
+    # missing docker-py handled in ansible.module_utils.docker.docker_common
     pass
 
 from ansible.module_utils._text import to_native
-from ansible.module_utils.docker_common import AnsibleDockerClient
+from ansible.module_utils.docker.docker_common import AnsibleDockerClient
```
Comment (WojciechowskiPiotr) [inline lib/ansible/module_utils/docker/docker_swarm.py]: How about `from ansible.module_utils.docker.common`? And same for swarm of course. As I see most modules in `module_utils` does not repeat the package name

**Comment by felixfontein** [comment]
This PR is mostly a error message / documentation PR, since it changes all Python 2.6 / docker-py related messages. Besides that, it moves the docker_* module_utils around, so we shouldn't keep it unmerged too long :-)

ready_for_review

**Review by WojciechowskiPiotr — COMMENTED** [review]
> This PR is mostly a error message / documentation PR, since it changes all Python 2.6 / docker-py related messages. Besides that, it moves the docker_* module_utils around, so we shouldn't keep it unmerged too long :-)
> 
> ready_for_review

@felixfontein - did you see my inline remark on naming submodules as `ansible.module_utils.docker.common` and `ansible.module_utils.docker.swarm`?

File: lib/ansible/module_utils/docker/docker_swarm.py (hunk 1 above)
Comment (felixfontein) [reply lib/ansible/module_utils/docker/docker_swarm.py]: Now that you actually posted the comment, I can see it :)

I'll change that. I've already thought about that, but then wanted to check something else first and then forgot about it...

**Comment by felixfontein** [comment]
I misunderstood something yesterday; Python 2.6 is still supported. I'll modify the PR this evening.

File: lib/ansible/module_utils/docker/docker_swarm.py (hunk 1 above)
Comment (WojciechowskiPiotr) [reply lib/ansible/module_utils/docker/docker_swarm.py]: I thought that inline reviews are visible straight away, hmm... Good we agree that this shorter naming is better :)

Comment (felixfontein) [reply lib/ansible/module_utils/docker/docker_swarm.py]: There are two buttons for each comment in review mode, one publishes them at that moment, and the other one waits until you finish the review. You're not the first one to accidentally holding back comments, though ;)

**Comment by felixfontein** [comment]
I've cleaned the requirement sections up, and did some refactoring using a cool feature of doc fragments I just learned about today :) (thanks @gundalow!)

(In case anyone wonders, the `options: {}` is needed as otherwise docs building fails...)

**Comment by felixfontein** [comment]
ready_for_review

**Comment by WojciechowskiPiotr** [comment]
Changing in some places exception handling to `except Exception as dummy` is part of some general guideline for coding?

**Comment by felixfontein** [comment]
For Ansible, `dummy` must be used for values which are never used (as opposed to `_` which is normally used in Python). So if the exception isn't used, it should be named `dummy`.
(https://docs.ansible.com/ansible/latest/dev_guide/testing/sanity/no-underscore-variable.html)

**Comment by WojciechowskiPiotr** [comment]
shipit

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Great work everybody.
Nice to see a PR which deletes more lines than it adds.

**Comment by felixfontein** [comment]
Thanks everyone!

**Pull request merged**